### PR TITLE
Feat line ending

### LINKED ISSUE #150: How should we reliably handle line ending
I don't think this is something we need to fix urgently, but I want to bring it up.

So, we don't want to force EOL to LF inside `.editorconfig` because this always conflict with git line ending normalization and mess up some windows tool. I know it's shitty, but I've been there and handling this while you work on windows is a nightmare. Let's be responsible and fix it correctly.

In almost any cases, we don't care whether we have LF or CRLF. The only case where it does matter is for `cli.js` because CRLF will break bash script.

I'd like us to have a foolproof prepublish hook that will ensure us correct package publication.

### PR DESCRIPTION
**Pull request by zckrs**
fix #150

Prepublish script will convert line ending to LF.

### CODE REVIEW

**Comment by hemanth** [comment]
LGTM :+1:

File: generators/gulp/templates/gulpfile.js [hunk 1]
```diff
@@ -34,6 +35,13 @@ gulp.task('nsp', function (cb) {
   nsp({package: path.resolve('package.json')}, cb);
 });
 
+gulp.task('lec', function () {
```
Comment (SBoudrias) [inline generators/gulp/templates/gulpfile.js]: Let's give a descriptive name

File: generators/gulp/templates/gulpfile.js [hunk 2]
```diff
@@ -34,6 +35,13 @@ gulp.task('nsp', function (cb) {
...
+  return gulp.src('**/*')
```
Comment (SBoudrias) [inline generators/gulp/templates/gulpfile.js]: We want to only force LF on cli files.

Comment (SBoudrias) [inline generators/gulp/templates/gulpfile.js]: Actually, as this is in pre-commit, it doesn't matter that much. But it'd still be better to only have cli file modified for windows users.

**Comment by hemanth** [comment]
ping @zckrs

**Comment by SBoudrias** [comment]
ping @zckrs hey, I'm still interested in getting this merged.

The only thing missing is to only apply this step to the _cli_ file.

**Comment by zckrs** [comment]
Hi, sorry for delay 
I will try take time for this today or tommorow.

**Comment by zckrs** [comment]
ping @SBoudrias 
Done. :smiley:

**Comment by SBoudrias** [comment]
Awesome! thanks :D

**Comment by hemanth** [comment]
:+1:

**Pull request merged**